### PR TITLE
First autotests for tests generation on Spring projects

### PR DESCRIPTION
**Pull request by EgorkaKulikov**
## Description

Try to make world better with Spring autotesting.

### Automated tests

Run utbot-samples pipeline and add these new tests to CI in future

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [x] The functionality I've repaired, changed or added is covered with **automated tests**.
- [ ] **Manual tests** have been provided optionally.
- [ ] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-framework-test/build.gradle [hunk 1]
```diff
@@ -8,6 +8,7 @@ dependencies {
 
     implementation(project(":utbot-framework"))
     testImplementation project(':utbot-sample')
+    testImplementation project(':utbot-spring-sample')
```
Comment (Damtev) [inline utbot-framework-test/build.gradle:11]: Why?

File: utbot-spring-test/build.gradle [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    testImplementation group: 'org.mockito', name: 'mockito-inline', version: mockitoInlineVersion
+    testImplementation group: 'org.jacoco', name: 'org.jacoco.report', version: jacocoVersion
+
+    testImplementation group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa', version: springBootVersion
+}
+
+configurations {
+    all {
+        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-logging'
+    }
```
Comment (Damtev) [inline utbot-spring-test/build.gradle:24]: Add a comment to this, please

File: utbot-spring-test/src/test/kotlin/org/utbot/examples/spring/autowiring/OrderServiceTests.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    configurations = standardSpringTestingConfigurations
+) {
+    @Test
+    fun testGetOrders() {
+        checkMocks(
+            method = OrderService::getOrders,
+            branches = eq(1),
+            { mocks, r ->
+                val orderRepository = mocks.singleMock("orderRepository", findAllRepositoryCall)
+                orderRepository.value<List<Order>?>(0) == r
```
Comment (Damtev) [inline utbot-spring-test/src/test/kotlin/org/utbot/examples/spring/autowiring/OrderServiceTests.kt:24]: Here and in all such cases - It's more readable to use `value` without arguments as it has a default `0` argument, I guess

**Review by Damtev — APPROVED** [review]
LGTM

File: utbot-testing/src/main/kotlin/org/utbot/testing/TestCodeGeneratorPipeline.kt [hunk 4]
```diff
@@ -86,13 +86,13 @@ class TestCodeGeneratorPipeline(private val testInfrastructureConfiguration: Tes
                     val prefix = when (codegenLanguage) {
                         CodegenLanguage.JAVA ->
                             when (parametrizedTestSource) {
-                                ParametrizedTestSource.DO_NOT_PARAMETRIZE -> "public void "
+                                ParametrizedTestSource.DO_NOT_PARAMETRIZE -> "@Test"
```
[the remark this answers is not in the document]
Comment (EgorkaKulikov) [reply utbot-testing/src/main/kotlin/org/utbot/testing/TestCodeGeneratorPipeline.kt:89]: Because of `setUp` and `tearDown` methods

**Pull request merged**